models/articles.model.js

```diff
@@ -24,3 +24,32 @@ exports.selectArticles = () => {
       return result.rows;
     });
 };
+
+const checkArticleIdExists = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1;", [article_id])
+    .then((data) => {
+      return data.rows.length !== 0;
+    });
+};
+exports.selectCommentsByArticleId = (article_id) => {
+  return db
+    .query(
+      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;",
+      [article_id]
+    )
+    .then((data) => {
+      if (data.rows.length === 0) {
+        return checkArticleIdExists(article_id).then((idExists) => {
+          if (idExists) {
+            return data.rows;
+          }
```

Comment:
Be carefully with nesting .then blocks we shouldn't really be doing it, it can cause unexpected behaviour.

I would also ask, is it not worth doing this check before you do the SELECT from COMMENTS db call, that way if the article doesn't exist you have saved yourself a database call.